delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -0,0 +1,465 @@
+                    health = data.get('health', '')
+                    status = constants.StoragePoolStatus.OFFLINE
+                    if health == 'OK':
+                        status = constants.StoragePoolStatus.NORMAL
+                    cpu_info = str(data.get('sc-cpu-type', ''))
+                    memory_size = data.get('system-memory-size') + "MB"
+                    system_memory_size = self.parse_string(
+                        memory_size)
+                    data_map = {
+                        'native_controller_id': data.get('controller-id', ''),
+                        'name': data.get('durable-id', ''),
+                        'storage_id': storage_id,
+                        'status': status,
+                        'location': data.get('position', ''),
+                        'soft_version': data.get('sc-fw', ''),
+                        'cpu_info': cpu_info,
+                        'memory_size': int(system_memory_size)
+                    }
+                    controller_arr.append(data_map)
+            return controller_arr
+        except Exception as e:
+            err_msg = "Failed to get storage controllers: %s"\
+                      % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def list_storage_volume(self, storage_id):
+        try:
+            volume_infos = self.ssh_pool.do_exec('show volumes')
+            volume_detail = self.handle_detail(volume_infos, 'volumes')
+            pools_info = self.ssh_pool.do_exec('show pools')
+            pool_detail = self.handle_detail(pools_info, 'pools')
+            list_volumes = []
+            if volume_detail:
+                for data in volume_detail:
+                    health = data.get('health', '')
+                    status = constants.StoragePoolStatus.OFFLINE
+                    if health == 'OK':
+                        status = constants.StoragePoolStatus.NORMAL
+                    total_size = self.parse_string(data.get('total-size'))
+                    total_avail = self.parse_string(data.get('allocated-size'))
+                    if pool_detail:
+                        native_storage_pool_id = pool_detail[0]. \
+                            get("serial-number")
+                        for pools in pool_detail:
+                            if data.get("virtual-disk-name") == pools.\
+                                    get("name"):
+                                native_storage_pool_id = pools.\
+                                    get("serial-number")
+                    data_map = {
+                        'name': data.get('volume-name', ''),
+                        'storage_id': storage_id,
+                        'description': data.get('volume-name', ''),
+                        'status': status,
+                        'native_volume_id': str(data.get('durable-id', '')),
+                        'native_storage_pool_id': native_storage_pool_id,
```

Comment:
`native_storage_pool_id` might be referenced before assignment.

Reply:
Defined in advance before use